.github/workflows/license_header_check.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import re
+import sys
+import yaml
+from pathlib import Path
+
+
+def recursive_glob(directories, pattern):
+    directory_paths = (Path(directory) for directory in directories)
+    files = (
+        directory_path.glob("**/" + pattern)
+        for directory_path in directory_paths
+    )
+    return list(files)
+
+
+def license_header_check(paths: list[str], headers, excludes):
+    print(
+        "------------------------- License Header Check Start -------------------------"
+    )
+    patterns = [re.compile(header) for header in headers]
+
+    has_files_without_header = False
+    files_without_header = 0
+    excluded_files = 0
+    scanned_files = 0
+    files_with_header = 0
+
+    for dir in recursive_glob(paths, "*"):
+        for file_path in dir:
+            if file_path.is_dir():
+                continue
+
+            scanned_files += 1
+
+            if any(file_path.match(pattern) for pattern in excludes):
+                excluded_files += 1
+                continue
+
+            with open(file_path, "r") as file:
+                content = file.read()
+                matches = [re.search(pattern, content) for pattern in patterns]
```

Comment:
The SPDX text must be in the first line of the file, you need to check only the first line.